core/src/abi.rs

```diff
@@ -82,22 +109,33 @@ impl ABIInput {
     pub fn generate_abi_name_properties(
         inputs: &[ABIInput],
         properties_type: &GenerateAbiPropertiesType,
-        prefix: Option<&str>,
-    ) -> Vec<GenerateAbiNamePropertiesResult> {
+        path: &Option<Vec<AbiNamePropertiesPath>>,
```

Comment:
`nitpick`: Slightly non-idomatic rust to pass a reference to an option, typically the option would simple passed plainly with an inner type reference, perhaps you could try this (or a vec if needed).

```suggestion
        path: Option<&[AbiNamePropertiesPath]>
```

Reply:
fixed